File: solutions/elixir/darts/1/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score(position) do
    cond do
      distance_square(position) > 100 -> 0
      distance_square(position) > 25 -> 1
      distance_square(position) > 1 -> 5
      true -> 10
    end
  end

  defp distance_square({x, y}), do: x*x + y*y
end

File: solutions/haskell/darts/1/src/Darts.hs
module Darts (score) where

distanceSquare :: Float -> Float -> Float
distanceSquare x y = x**2 + y**2

score :: Float -> Float -> Int
score x y
  | distanceSquare x y > 100 = 0
  | distanceSquare x y > 25 = 1
  | distanceSquare x y > 1 = 5
  | otherwise = 10
